src/msg/producer/ref_counted.go

```diff
@@ -51,8 +51,13 @@ func NewRefCountedMessage(m Message, fn OnFinalizeFn) *RefCountedMessage {
 }
 
 // Accept returns true if the message can be accepted by the filter.
-func (rm *RefCountedMessage) Accept(fn FilterFunc) bool {
-	return fn(rm.Message)
+func (rm *RefCountedMessage) Accept(fn []FilterFunc) bool {
```

Comment:
Maybe do a
```
if len(f) == 0 {
  return false
}
```
?

Reply:
This previously used to just be a single filter function, instead of a slice, so I don't believe it can be called with an empty slice. There's always a filter, the default filter which is the `acceptAllFilter`. 

I think we could introduce an error if the len(fn) == 0 because that would check if the registering/unregistering of filters led to an incorrect state where we have no filters.

Reply:
I see what you mean. But isn't that part of the caller (consumer service writer) ? This Accept() method looks like it is independent of that outer layer. In this instance you're right the csw always has one element in dataFilters but the producer/ref_counted.go has no control over its callers no? I haven't done a deep dive of the code so I could be wrong. I'll leave it up to you to decide. :-)

Reply:
Good point, I've gone ahead and added it.